src/modules/infobox/InfoboxModule.vue

```diff
@@ -58,14 +65,52 @@ function onPrint() {
     promptUserToPrintHtmlContent(content.value)
 }
 function onClose() {
-    store.dispatch('clearAllSelectedFeatures', dispatcher)
-    store.dispatch('clearClick', dispatcher)
+    if (showFeatureInfoInBottomPanel.value) {
+        store.dispatch('clearAllSelectedFeatures', dispatcher)
+        store.dispatch('clearClick', dispatcher)
+    } else if (showElevationProfile.value) {
+        // if feature details are shown in the floating tooltip we don't want to close the detail
+        // when clicking on the X button, but only close the profile
+        onHideProfile()
+    }
+}
+function onHideProfile() {
+    store.dispatch('setProfileFeature', { feature: null, ...dispatcher })
```

Comment:
I don't really like this way of doing things. If someone closes by mistake the profile, and open it again, they will make unnecessary calls to the backend. Now I understands the `ShowGeometryProfileButton` component sets the profile visibility by dispatching a `setProfileFeature` too, so it would require some extra modifications which might not be what you'd like to do in this PR, and would require extra work.


If there is a clean and quick way to do this, it could be great. Otherwise, it's okay as it is. (I don't think a lot of people are going to close and reopen their profile a lot. Maybe twice maximum per person ^^)

Reply:
Regarding call to backends : IMHO out of scope of this PR, and if our caching works properly calling twice the same request will hit our backend only once.

Regarding UI :
That is what was wanted for mobile use (which is the purpose of the infobox), do not have the profile AND the feature detail at the same time because it takes to much space